Make building tests explicitly optional

tmadlener opened the pull request:
Make building the tests explicitly optional by using `CTest` facilities, instead of having it implicitly depend on the presence of `GTest`.

This will add a new option `BUILD_TESTING` to the cmake configuration, which defaults to `ON`. If this option is set, then finding `GTest` is mandatory. It also enables integration with `ctest`, so that calling `ctest` after building will automatically run all tests (the original `bin/gtests` is still available and unchanged).

eckerpatrick commented:
I really like this solution, thanks for taking care of this!

tmadlener commented:
I realized that there are other tests as well, but not all of them are part of the usual CI run?
https://github.com/GenFit/GenFit/blob/a3e5ac1bf67176c5ff7d2e6e054c0c76cbcd625f/CMakeLists.txt#L310-L321
https://github.com/GenFit/GenFit/blob/a3e5ac1bf67176c5ff7d2e6e054c0c76cbcd625f/.travis.yml#L45-L47

I am not sure how to best handle those. Any preference from your side?

eckerpatrick commented:
Regarding the tests which are not included in the CI: I suppose the reason for this is that for them seperate windows are opened to have a look into the events and I am not sure how the CI would handle these. So I suppose they were mainly included for testing by hand.
About the streamerTest and unitTests: Is it possible to add them to the CTests?

tmadlener commented:
Ah OK, that makes sense. I have now added the two tests to the ones that are run automatically via `ctest` (plus an additional dummy test to first create the geometry). I have also simplified the travis config accordingly, as now everything is handled by CTest.

In summary: By default `BUILD_TESTING` is enabled and **all** tests are built regardless of whether they are added to the CTest suite or not. The tests that should run in CI can be triggered with `ctest` after the build. If `BUILD_TESTING` is disabled, the old behavior is restored and the `tests` target has to be built manually to get the test executables. If `BUILD_TESTING` is enabled, GTest becomes a mandatory dependency.

eckerpatrick commented:
This makes it easier to run the tests. Thanks for making the changes.
